src/engine/engine.h

```diff
@@ -6,4 +6,28 @@
 namespace mixxx {
     // TODO(XXX): When we move from stereo to multi-channel this needs updating.
     static constexpr mixxx::AudioSignal::ChannelCount kEngineChannelCount(2);
+
+    // Contains the information needed to process a buffer of audio
+    class AudioParameters : public AudioSignal {
```

Comment:
I'm not convinced about the naming. This class contains parameters for a single processing cycle within the engine.

EngineProcessParams/-Config/-Settings/-Context
EngineProcessingParams/...
...?

Reply:
What about EngineParameters?